### PR TITLE
Implements scipy.stats.rankdata

### LINKED ISSUE #1588: Support scipy.stats.rankdata
**Is your feature request related to a problem? Please describe.**
We've implemented Pearson coefficient with #1587. As Spearman coefficient is Pearson coefficient with rankings, we need ``scipy.stats.rankdata`` to compute rankings, and then Spearman coefficient can be worked out.

### PR DESCRIPTION
**Pull request by shantam-8**
## What do these changes do?

 Implements  scipy.stats.rankdata

## Related issue number

Fixes #1588

## Check code requirements

- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

File: mars/tensor/stats/tests/test_stats_execution.py [hunk 1]
```diff
@@ -139,29 +141,25 @@ def test_t_test_execution(setup):
...
-            )
+        sp_from_stats = lambda a, b, alternative=None, equal_var=True: sp_ttest_ind_from_stats(
+            a.mean(),
+            a.std(),
+            a.shape[0],
+            b.mean(),
+            b.std(),
+            b.shape[0],
+            alternative=alternative,
+            equal_var=equal_var,
```
Comment (zhongchun) [inline mars/tensor/stats/tests/test_stats_execution.py:162]: There is some failure of code style. Could you format this file with `black`?

Comment (shantam-8) [reply mars/tensor/stats/tests/test_stats_execution.py:162]: The following changes were made when I had run ``black`` in the first place. Should I ignore the changes made by ``black`` and manually restore them back to the original ones?

Comment (zhongchun) [reply mars/tensor/stats/tests/test_stats_execution.py:162]: You should upgrade `black` to the latest version.

File: mars/tensor/stats/rankdata.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    """
+    if method not in ("average", "min", "max", "dense", "ordinal"):
+        raise ValueError('unknown method "{0}"'.format(method))
+
+    if axis is not None:
+        a = np.asarray(a)
+        if a.size == 0:
+            np.core.multiarray.normalize_axis_index(axis, a.ndim)
+            dt = np.float64 if method == "average" else np.int_
+            return mt.empty(a.shape, dtype=dt)
```
Comment (zhongchun) [inline mars/tensor/stats/rankdata.py:84]: Could you add some tests to increase coverage?

File: mars/tensor/stats/rankdata.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    array([ 1,  2,  4,  3])
+    >>> rankdata([[0, 2], [3, 2]]).reshape(2,2).execute()
+    array([[1. , 2.5],
+          [4. , 2.5]])
+    >>> rankdata([[0, 2, 2], [3, 2, 5]], axis=1).execute()
+    array([[1. , 2.5, 2.5],
+           [2. , 1. , 3. ]])
+    """
+    if method not in ("average", "min", "max", "dense", "ordinal"):
+        raise ValueError('unknown method "{0}"'.format(method))
```
Comment (zhongchun) [inline mars/tensor/stats/rankdata.py:77]: This line is not covered yet. Could you add a test to cover it?

Comment (shantam-8) [reply mars/tensor/stats/rankdata.py:77]: I did notice that but I was not sure how to go about it. Could you please provide me with some help regarding that?

Comment (zhongchun) [reply mars/tensor/stats/rankdata.py:77]: We can set a method value other than `"average", "min", "max", "dense", "ordinal"` and use `with pytest.raises(ValueError)` to catch the expected exception. Or just add `# pragma: no cover` after line 76 to ignore it like https://github.com/mars-project/mars/blob/master/mars/core/entity/tileables.py#L90

File: mars/tensor/stats/__init__.py [hunk 4]
```diff
@@ -17,3 +17,4 @@
 from .ks import ks_1samp, ks_2samp
 from .power_divergence import power_divergence
 from .ttest import ttest_1samp, ttest_ind, ttest_ind_from_stats, ttest_rel
+from .rankdata import rankdata
```
Comment (hekaisheng) [inline mars/tensor/stats/__init__.py:20]: We sort imports alphabetically, so better to move this line up one line.

**Review by zhongchun — APPROVED** [review]
LGTM.

**Review by chaokunyang — APPROVED** [review]
LGTM, thanks for your contribution

**Pull request merged**